src/coreclr/debug/ee/controller.cpp

```diff
@@ -9054,6 +9054,23 @@ bool DebuggerDataBreakpoint::TriggerSingleStep(Thread *thread, const BYTE *ip)
         this->DisableSingleStep();
         return true;
     }
+    else if (g_pDebugger->m_isSuspendedForGarbageCollection)
+    {
+        // The debugger is not interested in Data Breakpoints during garbage collection
+        // We can safely ignore them since the Data Breakpoints are now on pinned objects
+        LOG((LF_CORDB, LL_INFO10000, "D:DDBP: Ignoring data breakpoint while in SuspendEE() for GC \n"));
+
+        // We enabled single step in TriggerDataBreakpoint so that the ScanForTriggers
```

Comment:
Refers to https://github.com/dotnet/runtime/blob/09a1e77f440b478e212195f0d6a4c8fa2564e825/src/coreclr/debug/ee/controller.cpp#L9020-L9022 above